src/Libraries/Storage/UploadedFile.php

```diff
@@ -130,9 +129,22 @@ public function __construct(array $meta)
         $this->originalName = $meta['name'];
         $this->errorCode = $meta['error'];
 
+        $this->loadAllowedMimeTypesFromConfig();
+
         parent::__construct($meta['tmp_name']);
     }
 
+    /**
+     * Overrides the default allowed mime types => extensions map
+     * @param array $allowedMimeTypes
+     * @return $this
+     */
+    public function setAllowedMimeTypes(array $allowedMimeTypes): UploadedFile
```

Comment:
These 2 methods `setAllowedMimeTypes()` and `loadAllowedMimeTypesFromConfig()` share common logic which may lead to maintenance issues.

Reply:
Also these methods are completely overrides the default values and does not extend it, need to support possibility to extend too like: 

`public function setAllowedMimeTypes(array $allowedMimeTypes, bool $merge = false)`